src/main/java/Parser.java

```diff
@@ -0,0 +1,164 @@
+                    } else if(itemANDtime[0].trim().isEmpty()) {
+                        throw new ArgumentException("Oh? Looks like your item disappeared into space.");
+                    } else if(itemANDtime[1].trim().isEmpty()) {
+                        throw new ArgumentException("Hiding from reality I see. Too bad time waits for no man");
+                    }
+
+                    String time[] = itemANDtime[1].split(" ", 2);
+
+                    if (!time[0].equals("by")) {
+                        throw new ArgumentException("Where did your BY go?");
+                    } else if (time[1].trim().isEmpty()) {
+                        throw new ArgumentException("A blank time I see");
+                    }
+                    String deadlineTime = itemANDtime[1].split(" ", 2)[1];
+
+                    SimpleDateFormat converterDate = new SimpleDateFormat("dd/MM/yyyy HH:mm");
+                    Date date = converterDate.parse(deadlineTime);
+
+                    System.out.println("Got it. I've added this task:");
+                    toDo.addItemDeadline("D", itemANDtime[0], date, deadlineTime);
+
+                } else {
+
+                    if(!item.contains("/from") || !item.contains("/to")) {
+                        throw new ArgumentException("For event please enter the format: event nameOfEvent /from dd/MM/yyyy HH:mm" +
+                                " /to dd/MM/yyyy HH:mm");
+                    }
+
+                    String itemANDtime[] = item.split(" /from ", 2);
+
+                    if (itemANDtime.length < 2) {
+                        throw new ArgumentException("You forget the event name!");
+                    } else if(itemANDtime[0].trim().isEmpty()) {
+                        throw new ArgumentException("Oh? Looks like your item disappeared into space.");
+                    }
+
+                    String nameItem = itemANDtime[0];
+
+                    String startEnd [] = itemANDtime[1].split(" /to ");
+
+                    if(startEnd.length <2) {
+                        throw new ArgumentException("Please enter the format: event nameOfEvent /from xxx /to xxx");
+                    }
+
+                    if(startEnd[0].trim().isEmpty() || startEnd[1].trim().isEmpty()) {
+                        throw new ArgumentException("Please do not leave the timing blank!");
+                    }
+
+                    SimpleDateFormat converterStartDate = new SimpleDateFormat("dd/MM/yyyy HH:mm");
+                    Date startDate = converterStartDate.parse(startEnd[0]);
+                    Date endDate =  converterStartDate.parse(startEnd[1]);
+
+
+                    System.out.println("Got it. I've added this task:");
+                    toDo.addItemEvent("E", nameItem, startDate, endDate, startEnd[0], startEnd[1]);
+                }
+
+                System.out.println("Now you have " + toDo.numberOfTask() + " tasks in the list.");
+
+            }
```

Comment:
Good job here! Another alternative might be to consider using switch to make the code more readable, but overall still good job :)